codegen/src/sbt-test/generator/e2e/src/test/scala/proto/test/MultiServiceSpec.scala

```diff
@@ -0,0 +1,32 @@
+package proto.test
+
+import com.twitter.util.{Await, Future}
+import com.soundcloud.twinagle.ServerBuilder
+
+import org.specs2.mutable.Specification
+
+class MultiServiceSpec extends Specification {
+  
+  val svc1 = new Service1Service {
+    override def rpc1(req: Request): Future[Response] = Future.value(Response())
+  }
+  
+  val svc2 = new Service2Service {
+    override def rpc2(req: Request): Future[Response] = Future.value(Response())
+  }
+
+
+
+  "ServerBuilder allows building services that support multiple Protobuf services" in {
+    val httpService = ServerBuilder()
+      .register(Service1Service.protoService(svc1))
```

Comment:
I'm not super happy here API-wise: it'd be a bit nicer for users not to have to mention the class explicitly. I didn't want to add the method to the `MyService` trait directly (e.g. `.register(svc1.asProtoService)`), since it's only relevant for the server-side usage, and the generated `MyClientProtobuf`/`MyClientJson` classes would end up with a weird method.

Even nicer would be `.register(svc1)`. To achieve that, we could introduce a typeclass (e.g. `AsProtoService[T]`) and generate an implementation of it in the `MyService` companion object. The `register` method would then have the signature `def register[T: AsProtoService](svc: T): ServerBuilder`. WDYT? Is it worth it?

Reply:
The 2nd option would be really elegant. And in a framework-y library, I'm often inclined to spend a lot of effort to achieve really client usage code.

I'm also curious to see the implementation.

Reply:
I gave it a shot. wdyt?